src/Parser/Shaman/Restoration/CombatLogParser.js

```diff
@@ -145,24 +146,29 @@ class CombatLogParser extends CoreCombatLogParser {
     const unbuffedHealingSurges = healingSurges - twHealingSurges;
     const totalTwUsed = twHealingWaves + twHealingSurges;
     const giftOfTheQueenCasts = giftOfTheQueen.casts || 0;
+    const giftOfTheQueenDuplicateCasts = giftOfTheQueen.casts || 0;
```

Comment:
What is the point of setting the cast event to the `Duplicate` casts also? This seems redundant and could be handled by knowing if the player has the trait or not.

Reply:
yup that's a typo i didn't catch, should be giftDuplicate.casts

Reply:
So just so you are aware - when I did a quick check on the logs, the `Duplicate` event didn't seem to have an associated Cast event. I would just double check to make sure that a cast event is actually generated so this number isn't always zero and therefore influences the below calculation.